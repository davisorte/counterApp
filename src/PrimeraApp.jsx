import PropTypes from 'prop-types'

//En la version 17 ya no es necesario importar react
//Utilizando desestructuracion en las props
const PrimeraApp=({name,titulo,subtitulo})=>{
    /**<> : no muestra el div en el navegador
     * {}: ver el contenido no imprime valores boleanos
     * se pueden muestrar areglos
     * {JSON.stringify(saludo)}
     * No se pueden imprimir objectos sin antes comvertirlos a json
     */
    /*const saludo={
        nombre:'DAvid',
        edad:19
    }*/
    //props : manda informacion entre los componentes
    return (
       
        <>
            {/*<h1>{JSON.stringify(saludo,null,3)}</h1>*/}
            <p>{name}</p>
            <h1 data-testid="test-title">{titulo}</h1>
            <p>{subtitulo}</p>
            <p>{subtitulo}</p>
        </>

        
    )
    
    
    
    
}
PrimeraApp.propTypes ={
    //va a recibir un valor de tip√≥ string
   titulo:PropTypes.string.isRequired,
   subtitulo:PropTypes.string.isRequired
}
//valor por defecto desde props
PrimeraApp.defaultProps={
    name:'David Orea Alvarez',
   // titulo:'No hay titulo',
    subtitulo:'No hay subtitulo'
}
export default PrimeraApp;