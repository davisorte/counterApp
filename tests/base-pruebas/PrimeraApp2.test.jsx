import { render , screen} from "@testing-library/react";
import PrimeraApp from "../../src/PrimeraApp";


describe('Pruebas en <PrimeraApp/>',()=>{
    const title ='Hola que hay bajo el agua'
    const subTitle ='subtitulo';
    //update snapshop; con la tecla u
    test('debe de mostrar el titulo en un p',()=>{
        
        const {container} = render(<PrimeraApp titulo={title}/>)
        expect(container).toMatchSnapshot();
        
    });
    test('debe de mostrar el "Hola que hay bajo el agua"',()=>{
        render(<PrimeraApp titulo={title}/>)

        //screen.debug(): muestra todo el html del componente
        //heading: 
        expect(screen.getByRole('heading',{level:1}).innerHTML).toContain( title );
    });

    render(<PrimeraApp 
        titulo={title}
        subtitulo ={subTitle}
        />)
    //toBeTruthy: pruebas flexibles debe tener un titlo
    expect(screen.getAllByText(subTitle).length).toBe(2);
});