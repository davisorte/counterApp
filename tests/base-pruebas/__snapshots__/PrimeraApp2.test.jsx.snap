// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp/> debe de mostrar el titulo en un p 1`] = `
<div>
  <p>
    David Orea Alvarez
  </p>
  <h1
    data-testid="test-title"
  >
    Hola que hay bajo el agua
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
</div>
`;
